### PR TITLE
Searched Stock can be saved in database, User-Repo implementet

### PR DESCRIPTION
**Pull request by VincAhr**
Security wird im Anschluss implementiert.

### CODE REVIEW

File: backend/src/main/java/com/example/demo/stock/ApiController.java [hunk 1]
```diff
@@ -9,12 +9,20 @@
 @Data
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/demo/stock/ApiController.java:9]:
```suggestion
@RequiredArgsConstructor
```

Overkill vor controller, only need the constructor

File: backend/src/main/java/com/example/demo/stock/StockData.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.example.demo.stock;
+
+
+import org.springframework.data.mongodb.core.mapping.Document;
+
+@lombok.Data
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/demo/stock/StockData.java:6]:
```suggestion

import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;

@Data
```

File: backend/src/main/java/com/example/demo/stock/StockService.java [hunk 3]
```diff
@@ -13,8 +13,13 @@
 public class StockService {
 
     @Value("${marketstack_pasword}") String pw;
+    private final StockRepository stockRepository;
 
-    public Stockdata searchStock (String stockName) {
+    public StockService(StockRepository stockRepository) {
+        this.stockRepository = stockRepository;
+    }
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/demo/stock/StockService.java:20]: Here you can also use the RequiredArgsConstructor

File: backend/src/main/java/com/example/demo/stock/StockService.java [hunk 4]
```diff
@@ -24,6 +29,8 @@ public Stockdata searchStock (String stockName) {
     }
 
 
-
-
+    public StockData addStock(StockData newStock) {
+        StockData saved = stockRepository.save(newStock);
+        return saved;
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/example/demo/stock/StockService.java:34]:
```suggestion
        return stockRepository.save(newStock);
```

**Review by Droggelbecher92 — CHANGES_REQUESTED** [review]
A bit chaotic :D Sometimes you use @RequiredArgsConstructor, sometimes you write the Constructor. Choose one, otherwise you'll get confused later on.

File: backend/src/main/java/com/example/demo/user/UserController.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.example.demo.user;
+
+public class UserController {
+}
```
[the remark this answers is not in the document]
Comment (VincAhr) [reply backend/src/main/java/com/example/demo/user/UserController.java:4]: Don't want the pull request to be too big, and for an usercontroller i thougt that i hat to implement the sucitiy stuff to make it complete. So i made a break here, but yes maybe your a right, it is not very logical to leave the controller empty.

**Pull request merged**